.github/workflows/integration_test.yml

```diff
@@ -1,8 +1,25 @@
 name: Integration Tests
 on: push
 jobs:
+  check-secret:
+    runs-on: ubuntu-latest
+    outputs:
+      secret-exists: ${{ steps.check-secret.outputs.defined }}
+    steps:
+      - name: Check for SLACK_API_TOKEN availability
+        id: check-secret
+        shell: bash
+        run: |
+          if [ "${{ secrets.SLACK_API_TOKEN }}" != '' ]; then
+            echo "defined=true" >> $GITHUB_OUTPUT;
+          else
+            echo "defined=false" >> $GITHUB_OUTPUT;
+          fi
+
   test:
     runs-on: ubuntu-latest
+    needs: [check-secret]
+    if: needs.check-secret.outputs.secret-exists == 'true'
```

Comment:
~Does a simple `if: ${{ secrets. SLACK_API_TOKEN }} != ''` not work? Do we need a whole extra step?~

I read up on this in https://stackoverflow.com/questions/70249519/how-to-check-if-a-secret-variable-is-empty-in-if-conditional-github-actions, sounds like this is the way :(